### PR TITLE
Add MAX Framework package

### PR DESCRIPTION
**Pull request by bdwyer2**

### CODE REVIEW

File: core/model.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+logger = logging.getLogger()
+
+
+class ModelWrapper(object):
+    """Model wrapper for Keras models"""
+
+    MODEL_NAME = 'resnet50'
+    DEFAULT_MODEL_PATH = 'assets/{}.h5'.format(MODEL_NAME)
+    MODEL_INPUT_IMG_SIZE = (224, 224)
```
Comment (djalova) [inline core/model.py]: I don't think this variable is being used

File: core/model.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import logging
+
+logger = logging.getLogger()
+
+
+class ModelWrapper(object):
+    """Model wrapper for Keras models"""
+
+    MODEL_NAME = 'resnet50'
+    DEFAULT_MODEL_PATH = 'assets/{}.h5'.format(MODEL_NAME)
```
Comment (djalova) [inline core/model.py]: This can stay in the config

File: core/model.py (hunk 1 above)
Comment (bdwyer2) [reply core/model.py]: Line 50

File: core/model.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        clear_session()
+        self.model = models.load_model(path)
+        # this seems to be required to make Keras models play nicely with threads
+        self.model._make_predict_function()
+        logger.info('Loaded model: {}'.format(self.model.name))
+
+    def read_image(self, image_data):
+        return Image.open(io.BytesIO(image_data))
+
+    def _preprocess_image(self, image, target, mode='tf'):
```
Comment (djalova) [inline core/model.py]: I think this can be refactored to be `_pre_process()`

File: core/model.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    def _pre_process(self, image, target, mode='tf'):
+        image = image.resize(target)
+        image = img_to_array(image)
+        image = np.expand_dims(image, axis=0)
+        return imagenet_utils.preprocess_input(image, mode=mode)
+
+    def _post_process_result(self, preds):
+        return imagenet_utils.decode_predictions(preds)[0]
+
+    def predict(self, x):
```
Comment (djalova) [inline core/model.py]: Can also refactor this to `_predict()` and `_post_process_result()` to `_post_process()`

File: core/model.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        clear_session()
+        self.model = models.load_model(path)
+        # this seems to be required to make Keras models play nicely with threads
+        self.model._make_predict_function()
+        logger.info('Loaded model: {}'.format(self.model.name))
+
+    def read_image(self, image_data):
+        return Image.open(io.BytesIO(image_data))
+
+    def _pre_process(self, image, target, mode='tf'):
```
Comment (djalova) [inline core/model.py]: The extra args will break the function in the base class, but I don't think we need target or mode.

File: core/model.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        image = img_to_array(image)
+        image = np.expand_dims(image, axis=0)
+        return imagenet_utils.preprocess_input(image, mode=self.MODEL_MODE)
+
+    def _post_process(self, preds):
+        return imagenet_utils.decode_predictions(preds)[0]
+
+    def _predict(self, x):
+        x = self._pre_process(x)
+        preds = self.model.predict(x)
```
Comment (djalova) [inline core/model.py]: I think we just need this line here.
The predict in the base class will call the other functions.

Comment (bdwyer2) [reply core/model.py]: Just tested locally and it doesn't seem like `_post_process()` is being called when I changed `def _predict()` to:
```python
    def _predict(self, x):
        x = self._pre_process(x)
        preds = self.model.predict(x)
        return preds
        # return self._post_process(preds)
```

File: core/model.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    MODEL_INPUT_IMG_SIZE = (224, 224)
+    MODEL_LICENSE = 'MIT'
+    MODEL_MODE = 'caffe'
+    MODEL_META_DATA = {
+        'id': '{}-keras-imagenet'.format(MODEL_NAME.lower()),
+        'name': '{} Keras Model'.format(MODEL_NAME),
+        'description': '{} Keras model trained on ImageNet'.format(MODEL_NAME),
+        'type': 'image_classification',
+        'license': '{}'.format(MODEL_LICENSE)
+    }
```
Comment (djalova) [inline core/model.py]: We can add `source` here

**Pull request merged**